lib/modulr/api/payments_service.rb

```diff
@@ -3,30 +3,55 @@
 module Modulr
   module API
     class PaymentsService < Service
-      def create( # rubocop:disable Metrics/ParameterLists
-        account_id:,
-        currency:,
-        amount:,
-        destination:,
-        reference:,
-        options: {}
-      )
-        data = {
+      def find(id:)
+        response = client.get("/payments", { id: id })
+        payment_attributes = response.body[:content]&.first
+        raise NotFoundError, "Payment #{id} not found" unless payment_attributes
+
+        Resources::Payments::Payment.new(response, payment_attributes)
+      end
+
+      def list(**opts)
+        return find(id: opts[:id]) if opts[:id]
+
+        response = client.get("/payments", build_query_params(opts))
+        Resources::Payments::Payments.new(response, response.body[:content])
+      end
+
+      # rubocop:disable Metrics/ParameterLists
+      def create(account_id:, destination:, reference:, currency:, amount:, **opts)
+        payload = {
           sourceAccountId: account_id,
+          destination: destination,
+          reference: reference,
           currency: currency,
           amount: amount,
-          reference: reference,
-          destination: {
-            type: destination[:type],
-            iban: destination[:iban],
-            name: destination[:name],
-          },
         }
-        data[:externalReference] = options[:external_reference] if options[:external_reference]
 
-        response = client.post("/payments", data, options)
+        payload[:externalReference] = opts[:external_reference] if opts[:external_reference]
+        payload[:endToEndReference] = opts[:e2e_reference] if opts[:e2e_reference]
+
+        response = client.post("/payments", payload)
         Resources::Payments::Payment.new(response, response.body)
       end
+      # rubocop:enable Metrics/ParameterLists
+
+      # rubocop:disable Metrics/AbcSize
+      private def build_query_params(opts)
+        same_name_params = [:type, :status]
+        date_params = { to: :toCreatedDate, from: :fromCreatedDate, updated_since: :modifiedSince }
+        mapped_params = {
+          external_reference: :externalReference,
+          has_external_reference: :hasExternalReference,
+          account_id: :sourceAccountId,
+        }
+        {}.tap do |params|
+          same_name_params.each { |param| params[param] = opts[param] if opts[param] }
+          date_params.each { |original, mapped| params[mapped] = format_datetime(opts[original]) if opts[original] }
+          mapped_params.each { |original, mapped| params[mapped] = opts[original] if opts[original] }
+        end
+      end
+      # rubocop:enable Metrics/AbcSize
```

Comment:
When developing a library, I usually favour ergonomics for the calling code, even if that involves some little extra work inside the library. Is that reasoning that guides me to propose this code:

```rb
client.payments.list(from: Date.today - 1)
```

instead of this one:

```rb
client.payments.list(fromCreatedDate: "2023-02-10T12:00:00+0000")
```

The tradeoff is obvious: there is a small indirection between the raw API (and what you see in the official documentation) and how one uses the library. 

We WERE already doing that when parsing the output. For example, we were mapping the responses of the API in a way that allows talking about `account.identifiers.first.country_details` instead of `account.identifiers.first.countrySpecificDetails`. This approach extends that rationale and applies to the data that is _provided_ to the API.

I understand that the tradeoff may not be tasteful to everybody in the team, so I'm happy to hear feedback on this from @devengoapp/dev. Just in case I made this piece disposable by just removing the method and changing:

```rb
response = client.get("/payments", build_query_params(opts))
```
to 

```rb
response = client.get("/payments", opts)
```

Reply:
I have to admit I don't like to read two documents (Modulr and our client one) to know what parameter names we can use. One thing we could make is to have good gem documentation where we can see all methods and available options and responses, so we can skip the Modulr documentation for most of the scenarios.

Reply:
In that case I'm in favor to make users life easier and not force them to pass arguments which client wrapper need to pass through thirdparty provider. I would pay the tradeoff to map this arguments in request as we already do with responses.